04/User.js

```diff
@@ -1,3 +1,24 @@
 export default class User {
+    constructor({ email, password }) {
+        this.email = this.getEmail(email)
+        this.password = this.getPassword(password)
+    }
```

Comment:
Taka forma powoduje błąd:

![image](https://github.com/devmentor-pl/practice-js-testing/assets/55095257/ed6bd851-d802-4890-9546-af661e983752)

Jeśli piszemy `getEmail()` to zazwyczaj oznacza pobierz email z wnętrza klasy/obiektu.
u Ciebie ta metoda dodatkowo sprawdza poprawność, co jest robione w metodzie set...

Mam wrażenie, że wszystko chciałaś zawrzeć w jednym miejscu, co niepotrzebnie utrudnia sam kod i jego napisanie.

Ja bym podszedł do tego tak:
```
export default class User {
    constructor({ email, password }) {
        this.setEmail(email);
        this.setPassword(password)
    }

    setEmail(email) {
        if (!email.includes('@')) {
            throw new Error('Email must include "@" sign')
        }

        this.email = email;
    }

    setPassword(password) {
        if (password.length < 8) {
            throw new Error('Password length must be greater than 8')
        }

        this.password = password;
    }

    getEmail() {
        return this.email;
    }

    getPassword() {
        return this.password
    }

    login() {
        if (!this.email.includes('devmentor.pl')) {
            return false
        } else return true
    }
}
```